### DISCUSSION TITLE
Edge Middleware (GA) use Edge Functions, but the latter are still in Beta?

### DISCUSSION THREAD

**Discussion by AaronBeaudoin** [vercel/vercel#8361] [General] [answered]
In the Build Output API, Edge Middleware seems to be implemented via Edge Functions. Quote from the description of the [`routes[].middlewarePath`](https://vercel.com/docs/build-output-api/v3#build-output-configuration/supported-properties/routes:~:text=to%20different%20routes.-,middlewarePath,-String) config option:

> A middleware name that matches an Edge Function path that should be invoked as middleware.

Edge Middleware is now GA according to the Vercel blog as well as the docs, but the underlying Edge Functions they appear to be using are still Beta. **It seems a bit unsafe to mark a product as GA while it still depends on a Beta product.** What is the expectation for Edge Functions, how they relate to Edge Middleware, and when we can expect to see GA for both?

**Comment by leerob** [comment, marked as answer]
Both Edge Middleware and Edge Functions use the underlying [Edge Runtime](https://vercel.com/blog/introducing-the-edge-runtime). Middleware is GA (stable) including the APIs, framework support, developer experience, and more. Edge Functions are not quite there yet, even though they use the same underlying runtime. We're actively working on this, though!

And thanks for the shout, we just updated the documentation here too:

> Path to an Edge Runtime function that should be invoked as Middleware.
